### ISSUE TITLE
Fallback scenario

### ISSUE THREAD

**Issue by crodnun** [Great-Hill-Corporation/quickBlocks#354]
As requested at #340 we open a separate issue to study the fallback scenario behaviour.

==============
Initial setup:
=============

Ubuntu local node NOT running ethereum.

============
Build steps done from scratch
============
git clone ..... 
mkdir build
cd build
cmake ../src
make        (it builds 100% OK)
cd ../bin

==========================
Testing fallback with getBlock command
==========================

0) Got to HOME and delete the .quickBlocks folder completely

   cd $HOME

   rm -rf .quickBlocks

1)  export FALLBACK=parity    (we define the fallback at environment)

2) Go to bin and run this command:

./getBlock 1001001     

(we retrieve the information from parity succesfully, xxxxx just means block info)

getBlock argc: 2 [1:1001001] 
getBlock 1001001 
{
  xxxxxxxxxxxxxxxxxx
  xxxxxxxxxxxxxxxxxxxxxxx
  xxxxxxxxx
}

3) After running it for the first time, I go to HOME to see what was created there (remember that I removed .quickBlocks before starting the test)

I can see that only this structure is created:

~.quickBlocks
~.quickBlocks/cache

4) If I run the same command (getBlock 1001001) a second time, now it creates the toml file with this content:

carlos@ubuntu16:~/.quickBlocks$ cat quickBlocks.toml
[settings]
blockCachePath=/home/carlos/.quickBlocks/cache/

5) No changes at speed are observed, no entries are inserted in caches, no more folders are created repeating this command many times.

**Comment by tjayrush** [comment]
Okay. I understand now. There's a couple of issues:

1. In order to reproduce this, the end user would have to do this:

    cmake ../src
    make

and then do this:

    cd ~
    rm -fR ./quickBlocks

This is totally out of the ordinary. If the user does this, I think it's their fault if the programs don't operate correctly, so I don't feel that we have to fix this problem (the problem being that the ~/.quickBlocks/quickBlocks.toml is created incompletely -- that is without the 'display strings.'

The second issue is

2. FALLBACK=parity should fail as the only valid FALLBACK is 'infura.' Please create an issue to track this.

3. Show me the values for 'xxxxx' My guess is that it shows a bunch of zeros (i.e. an empty block). If you do `FALLBACK=infura getBlock 1001001` you'll probably get a different result. When I do this command: `FALLBACK=parity getBlock 1001001 | wc` I get 99 lines. How many lines do you get?

****
**Summary:** The first problem is not our problem--if user deletes configuration they get what they deserve. The second problem needs an issue called something like FALLBACK should only take 'infura' value. The third issue is probably a bug if it reports an empty block. If it does, please create a separate issue.

Thanks for your perseverance. We're slowing, incrementally getting better.

**Comment by crodnun** [comment]
OK, the "parity" part was my typo :-). I have "infura" configured, so not an issue.

Regarding the third issue, it reports valid block data:

{
    "gasLimit": 3141592,
    "gasUsed": 84000,
    "hash": "0x0b4c6fb75ded4b90218cf0346b0885e442878f104e1b60bf75d5b6860eeacd53",
    "logsBloom": "0x0",
    "blockNumber": 1001001,
    "parentHash": "0x96810a6076e621e311a232468bfd3dcfac08f4803b255af0f00300f47981c10f",
    "timestamp": 1455421883,
    "transactions": [
{        
            "hash": "0x730724cb08a6eb17bf6b3296359d261570d343ea7944a17a9d7287d77900db08",
            "blockHash": "0x0b4c6fb75ded4b90218cf0346b0885e442878f104e1b60bf75d5b6860eeacd53",
            "blockNumber": 1001001,
            "transactionIndex": 0,
            "nonce": 9839,
            "timestamp": 1455421883,
            "from": "0x68795c4aa09d6f4ed3e5deddf8c2ad3049a601da",
            "to": "0x789f40d61bb6436254d5b5f9cbb8cefb76916d46",
            "value": 1174501130000000000,
            "gas": 90000,
            "gasPrice": 50000000000,
            "cumulativeGasUsed": 0,
            "input": "0x",
            "receipt": {
                "contractAddress": "0x0",
                "gasUsed": 21000,
                "logs": [],
                "logsBloom": "0x0"
            },
            "gasUsed": 21000
        },
{        
            "hash": "0xef2ea39c20ba09553b2f3cf02380406ac766039ca56612937eed5e7f3503fb3a",
            "blockHash": "0x0b4c6fb75ded4b90218cf0346b0885e442878f104e1b60bf75d5b6860eeacd53",
            "blockNumber": 1001001,
            "transactionIndex": 1,
            "nonce": 173040,
            "timestamp": 1455421883,
            "from": "0x2a65aca4d5fc5b5c859090a6c34d164135398226",
            "to": "0xd77ba7ef37f87d5246f35de898133e09c92ecaca",
            "value": 2148415350000000000,
            "gas": 90000,
            "gasPrice": 50000000000,
            "cumulativeGasUsed": 0,
            "input": "0x",
            "receipt": {
                "contractAddress": "0x0",
                "gasUsed": 21000,
                "logs": [],
                "logsBloom": "0x0"
            },
            "gasUsed": 21000
        },
{        
            "hash": "0x5352c80aa2073e21ce6c4aa5488c38455f3519955ece7dca5af3e326797bcc63",
            "blockHash": "0x0b4c6fb75ded4b90218cf0346b0885e442878f104e1b60bf75d5b6860eeacd53",
            "blockNumber": 1001001,
            "transactionIndex": 2,
            "nonce": 0,
            "timestamp": 1455421883,
            "from": "0x9d72354fda3e2bcb4d23d1cf9d0c0385d1ab08fd",
            "to": "0x8ce45c9dd072be275a9ab6c3736b086702d77413",
            "value": 38120000000000000000,
            "gas": 21000,
            "gasPrice": 50000000000,
            "cumulativeGasUsed": 0,
            "input": "0x",
            "receipt": {
                "contractAddress": "0x0",
                "gasUsed": 21000,
                "logs": [],
                "logsBloom": "0x0"
            },
            "gasUsed": 21000
        },
{        
            "hash": "0x060e4cf9fa8d34a8b423b5b3691b2541255ff7974ff16699e104edcfb63bd521",
            "blockHash": "0x0b4c6fb75ded4b90218cf0346b0885e442878f104e1b60bf75d5b6860eeacd53",
            "blockNumber": 1001001,
            "transactionIndex": 3,
            "nonce": 0,
            "timestamp": 1455421883,
            "from": "0x30d285941dfb8bbbe4ec72b02c8d166be01322d2",
            "to": "0x9b0a028eafdecde3afc0fd00b7937098388b7c8a",
            "value": 61848232647269079,
            "gas": 21000,
            "gasPrice": 50000000000,
            "cumulativeGasUsed": 0,
            "input": "0x",
            "receipt": {
                "contractAddress": "0x0",
                "gasUsed": 21000,
                "logs": [],
                "logsBloom": "0x0"
            },
            "gasUsed": 21000
        }
    ]
}

I think that this mess is because you wrote this at the beginning at #340 :

Local node >> not << running with a configured fallback and no cache (possible default)
-- The first time it runs is will be slow
-- The ./quickBlocks folder and the cache should be created
-- The second time it runs it should be significantly faster


And with "no cache" I understood that I could remove completely the .quickBlocks folder as you mentioned at the second point detailed above and then test what happens.

If it is an impossible scenario, we should forget this and ammend the instructions at #340

**Comment by tjayrush** [comment]
Okay. The FALLBACK=parity issue is a non-issue. Got it.

The data is identical to what I get on my Mac from the cache. So that's good.

I think we can ignore the issue where the user installs and then deletes the configuration folder. When I said "clear the cache" in #340, I think I meant remove ~/.quickBlocks/cache folder. Sorry I wasn't more clear.

I think you may close this issue if you wish.

**Comment by crodnun** [comment]
Closing
